plugins/org.eclipse.n4js.flowgraphs/src/org/eclipse/n4js/flowgraphs/analyses/EdgeGuideQueue.java

```diff
@@ -96,65 +86,50 @@ EdgeGuide removeFirst() {
 
 	/** @return an {@link Iterable} over all {@link EdgeGuide} in the queue */
 	Iterable<EdgeGuide> getIterator() {
-		return currEdgeGuides;
+		return edgeGuideQueue;
 	}
 
 	private boolean isJoinGroup(EdgeGuide eg1, EdgeGuide eg2) {
 		Node nextN1 = eg1.getNextNode();
 		Node nextN2 = eg2.getNextNode();
-		JumpToken pathContext1 = eg1.getEdge().finallyPathContext;
-		JumpToken pathContext2 = eg2.getEdge().finallyPathContext;
-		boolean isJoinGroup = nextN1 == nextN2 && pathContext1 == pathContext2;
+		boolean isJoinGroup = nextN1 == nextN2 && eg1.finallyContext.equals(eg2.finallyContext);
 		return isJoinGroup;
 	}
 
 	private int compareForRemoveFirst(EdgeGuide eg1, EdgeGuide eg2) {
 		ControlFlowEdge e1 = eg1.getEdge();
 		ControlFlowEdge e2 = eg2.getEdge();
-		Node nextNode1 = eg1.getNextNode();
-		Node nextNode2 = eg2.getNextNode();
-		ControlFlowElement cfe1 = nextNode1.getControlFlowElement();
-		ControlFlowElement cfe2 = nextNode1.getControlFlowElement();
 		ControlFlowType cft1 = e1.cfType;
 		ControlFlowType cft2 = e2.cfType;
 
 		return ComparisonChain.start()
 				.compare(eg1, eg2, EdgeGuideQueue::compareJoined)
-				.compare(cfe1, cfe2, EdgeGuideQueue::compareDepth)
-				.compare(e1, e2, this::compareVisited)
+				.compare(eg1, eg2, EdgeGuideQueue::compareDeadFlowContext)
 				.compare(cft1, cft2, EdgeGuideQueue::compareEdgeTypes)
 				.compare(eg1, eg2, EdgeGuideQueue::compareJoining)
-				.compare(eg1, eg2, EdgeGuideQueue::compareInternalPosition)
-				.compare(nextNode1.hashCode(), nextNode2.hashCode())
-				.compare(e1.hashCode(), e2.hashCode())
+				.compare(eg1, eg2, EdgeGuideQueue::compareASTPosition)
 				.result();
 	}
 
+	/** Prioritize joined edges. */
```

Comment:
Isn't it better to use the `Comparable` interface?

Reply:
This is a very specific comparison of two EdgeGuides. Hence, I chose to implement in a separate method and not within the EdgeGuige.